File: Tareas/Tarea-4/src/models/product.model.js
import mongoose from 'mongoose';

const productSchema = new mongoose.Schema({
  name: { type: String, required: true, trim: true, maxlength: 150 },
  description: { type: String },
  brand: { type: String },
  stock: { type: Number, default: 0, min: 0 }
}, { timestamps: true });

export default mongoose.model('Product', productSchema);

File: Trabajos-Grupales/Python/ApiREST - Tarea 4/app/models/product_model.py
import mongoengine as me
from datetime import datetime

class Product(me.Document):
    name = me.StringField(required=True, max_length=150)
    description = me.StringField()
    brand = me.StringField()
    stock = me.IntField(min_value=0, default=0)

    created_at = me.DateTimeField(db_field='createdAt', default=datetime.utcnow)
    updated_at = me.DateTimeField(db_field='updatedAt', default=datetime.utcnow)

    version = me.IntField(db_field='__v')

    meta = {
        "collection": "products",
        "indexes": [{"fields": ["name"]}],
    }

    def save(self, *args, **kwargs):
        self.updated_at = datetime.utcnow()
        return super().save(*args, **kwargs)
